app/models/rook.rb

```diff
@@ -1,3 +1,26 @@
 class Rook < Piece
 
+	def valid_move?(dest_x, dest_y)
+
+		#a move can not be a non-move
+    	return false if dest_x == self.x_position && dest_y == self.y_position
+
+		#no obstructions allowed
+		return false if is_obstructed?(dest_x, dest_y) == true
+
+		#only vertical or horizontal moves allowed
+		return false if (dest_x < self.x_position || dest_x > self.x_position) && (dest_y < self.y_position || dest_y > self.y_position)
+
+		#no vertical moves off the board
+		return false if dest_x < 1 || dest_x > 8
+    	
+    	#no horizontal moves off the board
+    	return false if dest_y < 1 || dest_y > 8
```

Comment:
@robgraphix1 I feel that lines 15 and 18 are probably unnecessary because they can probably be put somewhere else like in the piece controller so that all chess piece moves are validated similarly. What are your thoughts on where these lines could go?